smart_open/s3.py

```diff
@@ -478,18 +525,9 @@ def seek(self, offset, whence=constants.WHENCE_START):
 
         Returns the position after seeking."""
         logger.debug('seeking to offset: %r whence: %r', offset, whence)
-        if whence not in constants.WHENCE_CHOICES:
-            raise ValueError('invalid whence, expected one of %r' % constants.WHENCE_CHOICES)
 
-        if whence == constants.WHENCE_START:
-            new_position = offset
-        elif whence == constants.WHENCE_CURRENT:
-            new_position = self._current_pos + offset
-        else:
-            new_position = self._content_length + offset
-        new_position = smart_open.utils.clamp(new_position, 0, self._content_length)
-        self._current_pos = new_position
-        self._raw_reader.seek(new_position)
+        self._raw_reader._position = self._current_pos
```

Comment:
Why are you tweaking an internal attribute of the raw reader here prior to seeking? It seems like you're breaking encapsulation.

Reply:
The issue is that _current_pos and _raw_reader._position don't always have the same value because of buffering, so `WHENCE_CURRENT` doesn't work. Latest code uses an alternate approach to fix that:

```
        if whence == constants.WHENCE_CURRENT:
            whence = constants.WHENCE_START
            offset += self._current_pos
```

 Not sure that this is cleaner, but 🤷‍♂️.